rpc/src/rpc.rs

```diff
@@ -3478,7 +3476,8 @@ pub mod rpc_full {
                 .preflight_commitment
                 .map(|commitment| CommitmentConfig { commitment });
             let preflight_bank = &*meta.bank(preflight_commitment);
-            let transaction = sanitize_transaction(unsanitized_tx)?;
+            let finalized_bank = &*meta.bank(Some(CommitmentConfig::finalized()));
+            let transaction = sanitize_transaction(unsanitized_tx, finalized_bank)?;
```

Comment:
Why does this need a finalized Bank? I'm working through my setup using `solana-test-validator`, and it's a pain to have to wait for finalization after creating/extending the address-lookup account. Seems like preflight commitment should be sufficient? Am I missing something?

Reply:
My original intent was to ensure that address lookup table usage is as safe as possible. If address lookup tables are used with the "processed" commitment, it could be possible that the table is different when the transaction is actually confirmed in a different block. But after thinking a bit more, I think that preflight commitment is more intuitive and inherently reflects the accepted risk of the requester anyways so I'll update to that.